include/stdexec/execution.hpp

```diff
@@ -637,64 +721,16 @@ namespace stdexec {
       move_constructible<remove_cvref_t<_Sender>> &&
       constructible_from<remove_cvref_t<_Sender>, _Sender>;
 
-  /////////////////////////////////////////////////////////////////////////////
-  // get_attrs
-  namespace __attrs {
-    struct __empty_attrs {
-      using __t = __empty_attrs;
-      using __id = __empty_attrs;
-    };
-
-    struct get_attrs_t {
-      template <class _Sender>
-        requires tag_invocable<get_attrs_t, const _Sender&>
-        constexpr auto operator()(const _Sender& __sender) const
-          noexcept(nothrow_tag_invocable<get_attrs_t, const _Sender&>)
-          -> tag_invoke_result_t<get_attrs_t, const _Sender&> {
-          static_assert(queryable<tag_invoke_result_t<get_attrs_t, const _Sender&>>);
-          return tag_invoke(*this, __sender);
-        }
-
-      // NOT TO SPEC
-      // This overload is subsumed by the __sender constrained overload
-      // below, which is provided for backwards compatibility. We'll enable
-      // this one when removing the compatibility overload.
-#if 0
-      template <class _Sender>
-        requires (!tag_invocable<get_attrs_t, const _Sender&>) &&
-          __awaitable<_Sender, no_env_promise>
-        constexpr auto operator()(const _Sender& __sender) const
-          noexcept -> __empty_attrs {
-          return {};
-        }
-#endif
-
-      // NOT TO SPEC
-      // For backwards compatibility, get_attrs returns a reference
-      // to the passed in sender. This will be removed eventually in favor
-      // of all sender types defining get_attrs.
-      template <class _Sender>
-        requires (!tag_invocable<get_attrs_t, const _Sender&>) &&
-          __sender<_Sender, no_env>
-        constexpr auto operator()(const _Sender& __sender) const
-          noexcept -> const _Sender& {
-          return __sender;
-        }
-    };
-  } // namespace __attrs
-  inline constexpr __attrs::get_attrs_t get_attrs{};
-  using __attrs::get_attrs_t;
-  using __attrs::__empty_attrs;
-
   /////////////////////////////////////////////////////////////////////////////
   // [execution.senders]
   template <class _Sender, class _Env = no_env>
     concept sender =
       // NOT TO SPEC
       // The sender related concepts are temporarily "in flight" being
-      // upgraded from P2300R5 to the get_attrs aware version of P2300.
+      // upgraded from P2300R5 to the get_env / enable_sender aware version
+      // in P2300R7.
       requires (const remove_cvref_t<_Sender>& __sndr) {
-        { get_attrs(__sndr) } -> queryable;
+        { get_env(__sndr) } -> queryable;
       } &&
       __sender<_Sender, no_env> &&
```

Comment:
I'd like to drop this requirement checking with `no_env`. Basically, we want to know if something is an R5 sender _or_ an R7 sender. I think we can express that directly.

Reply:
I haven't addressed this yet